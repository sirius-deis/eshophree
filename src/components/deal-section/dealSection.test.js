import { render, screen } from "@testing-library/react";
import DealSection from "./dealSection";

jest.mock("../../hooks/useFetch", () => ({
  __esModule: true,
  default: jest.fn(),
}));


describe("DealSection component", () => {
  beforeEach(() => {
    jest.clearAllMocks()
  });
  it("should match snapshot", () => {
    const useFetch = require("../../hooks/useFetch").default;
    const mockProducts = [
      { id: 1, name: "Product 1", price: 100 },
      { id: 2, name: "Product 2", price: 200 },
    ];
    useFetch.mockReturnValue([mockProducts, false]);
    const { container } = render(<DealSection />);
    expect(container).toMatchSnapshot();
  });
  it("should render loading state initially", () => {
    const useFetch = require("../../hooks/useFetch").default
    useFetch.mockReturnValue([null, true])

    render(<DealSection/>)
    
    expect(screen.queryByText("Deal Of The Day")).not.toBeInTheDocument()
  })
});