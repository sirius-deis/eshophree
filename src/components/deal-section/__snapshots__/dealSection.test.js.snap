// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DealSection component should match snapshot 1`] = `
<div>
  <section
    class="sc-jJTsDX eWPprp"
  >
    <div
      class="sc-fFGjHI dwKuma"
    >
      <div
        class="sc-ewnqHT dLOUny"
      >
        <div
          style="display: flex; align-items: center; gap: 0.7rem;"
        >
          <svg
            color="var(--additional-color)"
            fill="currentColor"
            font-size="2rem"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M323.56 51.2c-20.8 19.3-39.58 39.59-56.22 59.97C240.08 73.62 206.28 35.53 168 0 69.74 91.17 0 209.96 0 281.6 0 408.85 100.29 512 224 512s224-103.15 224-230.4c0-53.27-51.98-163.14-124.44-230.4zm-19.47 340.65C282.43 407.01 255.72 416 226.86 416 154.71 416 96 368.26 96 290.75c0-38.61 24.31-72.63 72.79-130.75 6.93 7.98 98.83 125.34 98.83 125.34l58.63-66.88c4.14 6.85 7.91 13.55 11.27 19.97 27.35 52.19 15.81 118.97-33.43 153.42z"
            />
          </svg>
          <div
            class="sc-gTRrQi gPElot"
            color="additional-color"
          >
            Deal Of The Day
          </div>
        </div>
        <div>
          <button
            class="sc-gxYJeL sc-kPTPQs kCKvOC"
            disabled=""
          >
            ‹
          </button>
          <button
            class="sc-gxYJeL sc-ispOId kCKvOC"
            disabled=""
          >
            ›
          </button>
        </div>
      </div>
      <div
        class="sc-bgqQcB iPXChy"
        width="100"
      />
      <div
        style="display: flex; gap: 1rem;"
      >
        <div
          style="overflow: hidden;"
        >
          <div
            class="sc-dicizt cTmuUj"
            style="transform: translateX(100%);"
          />
        </div>
      </div>
    </div>
  </section>
</div>
`;
